src/AjaxDynamicDataTable.jsx

```diff
@@ -24,6 +24,14 @@ class AjaxDynamicDataTable extends Component {
         this.loadPage(1);
     }
 
+    componentDidUpdate({ params: prevParams }) {
+        const { params: currentParams } = this.props;
+
+        if (JSON.stringify(prevParams) !== JSON.stringify(currentParams)) {
+            this.loadPage(1);
+        }
+    }
```

Comment:
Perhaps this could be simplified to the following?

```jsx
    componentDidUpdate(prevProps) {
         if (JSON.stringify(prevProps.params) !== JSON.stringify(this.props.params)) {
            this.loadPage(1);
        }
    }
```

Destructering seems excessive since these values are only used once.

Reply:
Sure. Can do, just our boilerplate complains if not deconstructing so was trying to follow internal standards a bit more (although there's a decent amount of changes for this to happen).

Reply:
I'd keep it simple for now. 👍

I don't think we should add a bunch of extra syntax when there is no reuse.